newIDE/app/src/UI/SemiControlledTextField.js

```diff
@@ -50,90 +45,94 @@ type Props = {|
   style?: Object,
   rows?: number,
   rowsMax?: number,
-  autoFocus?: boolean,
+  autoFocus?: 'desktop' | 'desktopAndMobileDevices',
   endAdornment?: React.Node,
 |};
 
+export type SemiControlledTextFieldInterface = {|
+  focus: () => void,
+  forceSetValue: (text: string) => void,
+  forceSetSelection: (start: number, end: number) => void,
+  getInputNode: () => ?HTMLInputElement,
+|};
+
 /**
  * This component works like a material-ui TextField, except that
  * the value passed as props is not forced into the text field when the user
  * is typing. This is useful if the parent component can do modifications on the value:
  * the user won't be interrupted or have the value changed until he blurs the field.
  */
-export default class SemiControlledTextField extends React.Component<
+const SemiControlledTextField = React.forwardRef<
   Props,
-  State
-> {
-  state = {
-    focused: false,
-    text: null,
-  };
-
-  _field: ?TextField = null;
+  SemiControlledTextFieldInterface
+>((props, ref) => {
+  const [focused, setFocused] = React.useState<boolean>(false);
+  const [text, setText] = React.useState<?string>(null);
+  const textFieldRef = React.useRef<?TextFieldInterface>(null);
 
-  forceSetValue(text: string) {
-    this.setState({ text });
-  }
+  const forceSetValue = (text: string) => {
+    setText(text);
+  };
 
-  forceSetSelection(selectionStart: number, selectionEnd: number) {
-    const input = this.getInputNode();
+  const forceSetSelection = (selectionStart: number, selectionEnd: number) => {
+    const input = getInputNode();
     if (input) {
       input.selectionStart = selectionStart;
       input.selectionEnd = selectionEnd;
     }
-  }
+  };
 
-  focus() {
-    if (this._field) this._field.focus();
-  }
+  const focus = () => {
+    if (textFieldRef.current) textFieldRef.current.focus();
+  };
+
+  const getInputNode = (): ?HTMLInputElement => {
+    if (textFieldRef.current) return textFieldRef.current.getInputNode();
+  };
 
-  getInputNode(): ?HTMLInputElement {
-    if (this._field) return this._field.getInputNode();
-  }
+  React.useImperativeHandle(ref, () => ({
+    focus,
+    getInputNode,
+    forceSetSelection,
+    forceSetValue,
+  }));
 
-  render() {
-    const {
-      value,
-      onChange,
-      commitOnBlur,
-      onFocus,
-      onBlur,
-      type,
-      ...otherProps
-    } = this.props;
+  const {
+    value,
+    onChange,
+    commitOnBlur,
+    onFocus,
+    onBlur,
+    type,
+    ...otherProps
+  } = props;
 
-    return (
-      // $FlowFixMe
-      <TextField
-        {...otherProps}
-        type={type || 'text'}
-        ref={field => (this._field = field)}
-        value={this.state.focused ? this.state.text : value}
-        onFocus={event => {
-          this.setState({
-            focused: true,
-            text: this.props.value,
-          });
+  return (
+    // $FlowFixMe
```

Comment:
is this flowfixme still required?

Reply:
Yes, I did not have the courage to fix it